### ISSUE TITLE
Bring in a faster Transform stream 

### ISSUE THREAD

**Issue by mcollina** [nodejs/performance#147]
https://github.com/mcollina/syncthrough offers 3x the throughput of `Transform`. https://www.npmjs.com/package/minipass offer similar characteristics.

I think bringing in a _faster_ set of Streams that _enforce_ backpressure would speed up an area of Node.js that is showing its time.

**Comment by rluvaton** [comment]
I don't think we should have another stream implementation, this will be confusing and need to maintain multiple implementations for complex topic

**Comment by ronag** [comment]
There is room to further optimize `stream.Transform` to be close to synchthrough. To begin with we should avoid the double buffering and implement `stream.Transform` as a `stream.Readable` with `stream.Writable` compatible api.

**Comment by ronag** [comment]
Also rather than adding another stream api, I think we should add support for sync functions in pipeline, i.e.

```
stream.pipeline(src, val => val.toUpperCase(), res)
```

**Comment by mcollina** [comment]
> There is room to further optimize stream.Transform to be close to synchthrough. To begin with we should avoid the double buffering and implement stream.Transform as a stream.Readable with stream.Writable compatible API.

I would love this, whoever the world depends on `_readableState` and `_writableState`. If you have ideas, I'm all ears.

> Also rather than adding another stream api, I think we should add support for sync functions in pipeline

@ronag that would have almost the same effect. Maybe we could internalize SyncThrough or analog and use ti to support pipeline

**Comment by rluvaton** [comment]
@ronag I remember u had some pr to remove the double buffering but stopped, do you remember what blocked you?

**Comment by ronag** [comment]
> @ronag I remember u had some pr to remove the double buffering but stopped, do you remember what blocked you?

Time was a blocker.
